framework/decode/vulkan_replay_dump_resources_common.cpp

```diff
@@ -1668,173 +1900,1000 @@ VkResult DumpTopLevelAccelerationStructure(DumpedTopLevelAccelerationStructure&
+    }
+
+    return res;
+}
+
+struct PushConstantBlock
+{
+    VkDeviceAddress array_of_pointers;
+    VkDeviceAddress out_buffer;
+    uint32_t        count;
+};
+
+static VkResult CreateComputeResources(const graphics::VulkanDeviceTable& device_table,
+                                       VkDevice                           device,
+                                       VkPipeline*                        compute_ppl,
+                                       VkPipelineLayout*                  ppl_layout)
+{
+    GFXRECON_ASSERT(compute_ppl != nullptr);
+    GFXRECON_ASSERT(ppl_layout != nullptr);
+
+    // We'll be using push constants to upload the device addresses to the compute shader
+    const VkPushConstantRange        push_constant_range{ VK_SHADER_STAGE_COMPUTE_BIT, 0, sizeof(PushConstantBlock) };
+    const VkPipelineLayoutCreateInfo pipelineLayoutCI{
+        VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO, nullptr, 0, 0, nullptr, 1, &push_constant_range
+    };
+
+    VkResult res = device_table.CreatePipelineLayout(device, &pipelineLayoutCI, nullptr, ppl_layout);
+    if (res != VK_SUCCESS)
+    {
+        GFXRECON_LOG_WARNING("vkCreatePipelineLayout failed (%s)", util::ToString(res).c_str());
+        return res;
+    }
+
+    const VkShaderModuleCreateInfo sci = {
+        VK_STRUCTURE_TYPE_SHADER_MODULE_CREATE_INFO, nullptr, 0, sizeof(g_CompShaderMain), g_CompShaderMain
+    };
+    VkShaderModule compute_shader;
+    res = device_table.CreateShaderModule(device, &sci, nullptr, &compute_shader);
+    if (res != VK_SUCCESS)
+    {
+        device_table.DestroyPipelineLayout(device, *ppl_layout, nullptr);
+        GFXRECON_LOG_WARNING("vkCreateShaderModule failed (%s)", util::ToString(res).c_str());
+        return res;
+    }
+
+    const VkPipelineShaderStageCreateInfo stage_ci = { VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO,
+                                                       nullptr,
+                                                       0,
+                                                       VK_SHADER_STAGE_COMPUTE_BIT,
+                                                       compute_shader,
+                                                       "ComputeMain",
+                                                       nullptr };
+
+    const VkComputePipelineCreateInfo ppl_ci = {
+        VK_STRUCTURE_TYPE_COMPUTE_PIPELINE_CREATE_INFO, nullptr, 0, stage_ci, *ppl_layout, VK_NULL_HANDLE, 0
+    };
+
+    res = device_table.CreateComputePipelines(device, VK_NULL_HANDLE, 1, &ppl_ci, nullptr, compute_ppl);
+    if (res != VK_SUCCESS)
+    {
+        device_table.DestroyPipelineLayout(device, *ppl_layout, nullptr);
+        GFXRECON_LOG_WARNING("vkCreateComputePipelines failed (%s)", util::ToString(res).c_str());
+    }
+
+    device_table.DestroyShaderModule(device, compute_shader, nullptr);
+
+    return VK_SUCCESS;
+}
+
+VkResult AccelerationStructureDumpResourcesContext::CloneBuildAccelerationStructuresInputBuffers(
```

Comment:
These and other functions later in this commit are very long, so it's hard for me to trace execution and dependencies.  Are there opportunities to break out parts of these functions into smaller functions that make this easier to read and reason about?  Like maybe utilities for inserting buffer copies and barriers into commandbuffers, and maybe breaking out subfunctions for the geometry types?